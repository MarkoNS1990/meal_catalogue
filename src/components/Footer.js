import React from 'react'
import '../styles/Footer.css'

function Footer() {
    return (
        <footer>
           <p>Made with &hearts; by Marko Zecevic &copy; 2021.</p> 
        </footer>
    )
}

export default Footer
