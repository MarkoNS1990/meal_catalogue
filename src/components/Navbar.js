import React from 'react'
import {BrowserRouter,Link} from 'react-router-dom'

function Navbar() {
    return (
            <BrowserRouter>  
            <nav>
                <ul>
                    <div className="meal-logo">
                        <li><Link to='/'>Meal catalogue</Link></li>
                    </div>
                    <div className="list-item">
                        <Link to='/'>Home</Link>
                        <Link to='/random'>Get random meal</Link>
                        <Link to='/about'>About us</Link>
                    </div>
                </ul>
            </nav>
            </BrowserRouter> 
        
    )
}

export default Navbar
