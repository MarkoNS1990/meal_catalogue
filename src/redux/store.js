import {createStore,applyMiddleware} from 'redux';
import thunk from 'redux-thunk'
import mealReducer from './meal/mealReducer';

const store = createStore(mealReducer,applyMiddleware(thunk))

export default store