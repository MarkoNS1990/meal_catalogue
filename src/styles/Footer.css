footer {
  background-color: rgb(36, 35, 35);
  color: #fff;
  padding: 20px;
  text-align: center;
  clear: both;
  margin-top: auto;
}
