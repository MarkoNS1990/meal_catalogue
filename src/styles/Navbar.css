.my-navbar {
  background-color: rgb(194, 60, 60);
  color: #fff;
  padding: 20px;
  font-size: 1.3rem;
}

ul {
  list-style-type: none;
}

a {
  color: #fff;
  text-decoration: none;
  margin-right: 30px;
  font-family: "Courier New", Courier, monospace;
}

a:hover {
  color: rgb(46, 43, 43);
  transition: 0.2s ease-in;
}

.nav-list {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.MuiInputBase-input::placeholder {
  color: #fff !important;
}

.search-item {
  display: flex;
  align-items: center;
}

.search-icon {
  background-color: rgb(194, 60, 60);
  border: none;
  color: #fff;
}

.meal-logo {
  font-family: "Courier New", Courier, monospace;
}

@media only screen and (max-width: 768px) {
  .nav-list {
    flex-direction: column;
  }

  .list-item {
    margin: 10px 0;
  }
}
