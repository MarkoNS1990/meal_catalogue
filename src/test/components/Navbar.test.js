import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import {BrowserRouter} from 'react-router-dom'
import Footer from '../../components/Footer';
import store from '../../redux/store';
import Navbar from '../../components/Navbar';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<BrowserRouter><Provider store={store}><Navbar /></Provider></BrowserRouter>);
  expect(asFragment(<Provider store={store}><Navbar /></Provider>)).toMatchSnapshot();
});

it('displays correct first child', () => {
  const { container } = render(<BrowserRouter><Provider store={store}><Navbar /></Provider></BrowserRouter>);
  expect(container.firstChild).toMatchSnapshot();
});

it('negative check', () => {
  render(<BrowserRouter><Provider store={store}><Footer /></Provider></BrowserRouter>);
  expect(screen.queryByText('some text!')).not.toBeInTheDocument();
});