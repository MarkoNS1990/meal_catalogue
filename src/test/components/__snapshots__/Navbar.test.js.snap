// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct first child 1`] = `
<nav
  class="my-navbar"
>
  <ul
    class="nav-list"
  >
    <div
      class="meal-logo"
    >
      <a
        href="/"
      >
        Meal catalogue
      </a>
    </div>
    <div
      class="list-item"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/random"
      >
        Random meal
      </a>
      <a
        href="/about"
      >
        About us
      </a>
    </div>
    <div
      class="search-item"
    >
      <div
        class="search-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline"
      >
        <input
          class="MuiInputBase-input MuiInput-input"
          placeholder="search meals"
          type="text"
          value=""
        />
      </div>
    </div>
  </ul>
</nav>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="my-navbar"
  >
    <ul
      class="nav-list"
    >
      <div
        class="meal-logo"
      >
        <a
          href="/"
        >
          Meal catalogue
        </a>
      </div>
      <div
        class="list-item"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/random"
        >
          Random meal
        </a>
        <a
          href="/about"
        >
          About us
        </a>
      </div>
      <div
        class="search-item"
      >
        <div
          class="search-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline"
        >
          <input
            class="MuiInputBase-input MuiInput-input"
            placeholder="search meals"
            type="text"
            value=""
          />
        </div>
      </div>
    </ul>
  </nav>
</DocumentFragment>
`;
